AWSCore/Authentication/AWSCredentialsProvider.m

```diff
@@ -600,12 +609,14 @@ - (void)setUpWithRegionType:(AWSRegionType)regionType
     if (cancellationTokenSource.isCancellationRequested) {
         return [AWSTask cancelledTask];
     }
+    
+    __block AWSCredentials *credentials = self.internalCredentials.copy;
     // Returns cached credentials when all of the following conditions are true:
     // 1. The cached credentials are not nil.
     // 2. The credentials do not expire within 10 minutes.
-    if (self.internalCredentials
-        && [self.internalCredentials.expiration compare:[NSDate dateWithTimeIntervalSinceNow:10 * 60]] == NSOrderedDescending) {
-        return [AWSTask taskWithResult:self.internalCredentials];
+    if (credentials
+        && [credentials.expiration compare:[NSDate dateWithTimeIntervalSinceNow:10 * 60]] == NSOrderedDescending) {
```

Comment:
This could be done with a computed property which can be tested and make the context more clear. First, add the property declaration to the class extension.

```objc
@interface AWSCredentials()

@property (readonly) BOOL credentialsExpired;

@end
```

Then implement the method.

```objc
- (BOOL)credentialsExpired {
    return (self.credentials &&
        [self.credentials.expiration compare:[NSDate dateWithTimeIntervalSinceNow:10 * 60]] == NSOrderedDescending);
}
```

In the tests code the property will have to be declared again to make it accessible. It appears this property could be used in more that 1 place in this source file.

Reply:
Added the helper method